src/NovelRT.Interop/Graphics/NrtRGBAColour.cpp

```diff
@@ -14,132 +14,30 @@ extern "C"
     {
-        RGBAColour* newColour = new RGBAColour(r, g, b, a);
-
-        return reinterpret_cast<NrtRGBAColourHandle>(newColour);
-    }
-
-    int32_t Nrt_RGBAColour_getR(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        return colourConfigPtr->getR();
-    }
-
-    NrtResult Nrt_RGBAColour_setR(NrtRGBAColourHandle colourConfig, int32_t inputValue)
-    {
-        if (colourConfig == nullptr)
-        {
-            Nrt_setErrMsgIsNullptrInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        colourConfigPtr->setR(inputValue);
-
-        return NRT_SUCCESS;
-    }
-
-    int32_t Nrt_RGBAColour_getG(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        return colourConfigPtr->getG();
-    }
-
-    NrtResult Nrt_RGBAColour_setG(NrtRGBAColourHandle colourConfig, int32_t inputValue)
-    {
-        if (colourConfig == nullptr)
-        {
-            Nrt_setErrMsgIsNullptrInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        colourConfigPtr->setG(inputValue);
-
-        return NRT_SUCCESS;
-    }
-
-    int32_t Nrt_RGBAColour_getB(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        return colourConfigPtr->getB();
-    }
-
-    NrtResult Nrt_RGBAColour_setB(NrtRGBAColourHandle colourConfig, int32_t inputValue)
-    {
-        if (colourConfig == nullptr)
-        {
-            Nrt_setErrMsgIsNullptrInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        colourConfigPtr->setB(inputValue);
-
-        return NRT_SUCCESS;
-    }
-
-    int32_t Nrt_RGBAColour_getA(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        return colourConfigPtr->getA();
-    }
-
-    NrtResult Nrt_RGBAColour_setA(NrtRGBAColourHandle colourConfig, int32_t inputValue)
-    {
-        if (colourConfig == nullptr)
-        {
-            Nrt_setErrMsgIsNullptrInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
-        colourConfigPtr->setA(inputValue);
-
-        return NRT_SUCCESS;
-    }
-
-    float Nrt_RGBAColour_getRScalar(NrtRGBAColourHandle colourConfig)
-    {
-        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(colourConfig);
+        RGBAColour* colourConfigPtr = reinterpret_cast<RGBAColour*>(&colourConfig);
```

Comment:
I think this might be more performant if we dereferenced the result of `reinterpret_cast` and used it as a value.